copystructure.go

```diff
@@ -48,14 +49,25 @@ type walker struct {
 	vals        []reflect.Value
 	cs          []reflect.Value
 	ps          []bool
+	locks       []sync.Locker
 }
 
 func (w *walker) Enter(l reflectwalk.Location) error {
 	w.depth++
+	// this normally appends only once, but make sure we start with enough
+	// space to use depth as an index
+	for w.depth >= len(w.locks) {
+		w.locks = append(w.locks, nil)
+	}
```

Comment:
Sorry, I'm not sure I follow. This is the only place `w.locks` is extended to match `w.depth` as we walk deeper, so it would still need to be append at some point. All other access is indexed directly.